ocaml/src/asd_server.ml

```diff
@@ -295,6 +295,47 @@ let get_value_option kv key =
 
 let key_exists kv key = (get_value_option kv key) <> None
 
+module Net_fd = struct
+  include Net_fd
+  (*
+    we don't want this to end up in the arakoon plugins,
+    where it's needed nor used
+    and pulls along cstruct and friends
+   *)
+  let sendfile_all ~fd_in ~(fd_out:t) size =
+    match fd_out with
+    | Plain fd ->
+       Fsutil.sendfile_all
+         ~fd_in
+         ~fd_out:fd
+         size
+    | SSL(_,socket) ->
+       let buffer = Lwt_bytes.create 4096 in
+       let copy_using buffer =
+         let buffer_size = Lwt_bytes.length buffer in
+          let write_all socket buffer offset length =
+            let write_from_source = Lwt_ssl.write_bytes socket buffer in
+            Lwt_extra2._write_all write_from_source offset length
+          in
+          let rec loop todo =
+            if todo = 0
+            then Lwt.return_unit
+            else
+              begin
+                let step =
+                  if todo <= buffer_size
+                  then todo
+                  else buffer_size
+                in
+                Lwt_bytes.read fd_in        buffer 0 step >>= fun bytes_read ->
+                write_all socket buffer 0 bytes_read >>= fun () ->
+                loop (todo - bytes_read)
+              end
+          in
+          loop size
+       in
+       copy_using buffer
```

Comment:
wrap this with `Lwt.finalize` to `Lwt_bytes.unsafe_destroy buffer`
(or maybe add `let Lwt_bytes.with_buffer length f = ...` and use that?)